### DISCUSSION TITLE
Requesting Contact to help contribute to expanding on data.

### DISCUSSION THREAD

**Discussion by sleepinonai** [kyleskom/NBA-Machine-Learning-Sports-Betting#411] [General]
Good afternoon,
First off I wanted to say thank you for this beautiful project you have put together.  I have been playing with it for the past couple of days.  I noticed, going through the discussions and feedback that you are trying to add player data.  I have a separate project that I would love to share with you which will pull the active roster for each team, starters for each team and an injury/illness report for each team.  This is all pulled through the nba api with no paywall or rate limit.  

Also I created a basketball-reference-scraper that scrapes all of the game logs for each player for each date, including advanced statistics.  This works through json responses.  I can also provide pinnacle odds for NBA games and player props, as well as Draft King Odds.  All of this is accomplished with no paywall, no rate limiting and should provide all of the information that you will need to continue to expand on this project.  

I am so happy to have stumbled across this and would love to share my code with you to help you pull these statistics and use them to help expand; as you have shared this beautiful project as well with us.  

My email address is: sleepinonai@gmail.com 

If you ever would like to reach out, I would love to share the resources that I have with you.  

May you continue to grow in knowledge!

Thank you~

**Comment by STRATZ-Ken** [comment]
This is an open source project. If you want to commit to it, feel free to do so. There should be no reason for people to contact you directly.  @kyleskom Delete this spam.

**Reply by sleepinonai** [reply]
My apologies. My purpose was never to spam but to engage with like minded individuals who don't want to sit behind a data provider to pull game logs, advanced stats, and odds.  I figured posting this in Discussion, under General, would be open to just that; an open discussion with the purpose of trying to provide insight to data points that can be used in the future with someone who is much more knowledgeable at implementing these data points into an already, well structured project - such as this.

**Reply by nealmick** [reply]
only questionable thing is the email, but people reply from email all the time.  Seems genuinely interested.  I just warn anyone if they doing this for the money printer your gona be disappointed.  Pretty much all casinos limit for profit.  So if you win consistently, you will be kicked out or limited.  this project is good for learning how machine learning models work and how to get and form data.  Other then that if you want to make money off sports betting, lookup arbitrage sports betting, or value betting market consensus.  Pretty standard stuff it isnt effortless but can make some amount before you get limited.  i have heard some people have been limited after just 1 or 2 bets.   Bookies analyze your betting patterns against other people to find advantaged players and limit or ban.  Casinos are private business and reserve the right to refuse you service for any reason including profit.

**Reply by STRATZ-Ken** [reply]
Correct, if you here to make money just go Arbitrage betting. If you here to learn, go for it, its a great project.

**Reply by sleepinonai** [reply]
Absolutely Agreed!  I absolutely love the structure of this project and even when rebuilding and running the training models through additional tests to measure the validation accuracy it was quite high.  So I examined the sql dataset and noticed that it wasn't populating the Team.1 stats at all which would represent the opponents OREB DREB etc for more feature building.  So I rewrote the Data pull to identify the opponents by comparing the Team and Date playing in the original dataset.sqlite to the Date and teams playing in the odds sql file to identify each opponent.  

Now its just feature building such as adding elo ratings for each season for each team, basic feature building....See if we can reconstruct the models with better features and getting the val_loss down.  I would love to see a .1 Val loss rather than a .6

So for everyone turning to this project thinking that your going to be rich....It's a stepping stone for building a strong model, beautifully structured, but you will have to manipulate the data a bit and begin building your own features...

Best of luck everyone

**Reply by nealmick** [reply]
I would re-evaluate your goal of building a strong model. basically same as money printer.   probably convince yourself its not for money, then be disappointed when dont make money....

**Comment by sleepinonai** [comment]
Also another interesting observation, while running the training for the models in an isolated Fedora container on my server; I was able to achieve much higher accuracy for the XGB Models rather than running within Windows through PyCharm.  I have spent much more time attempting to train on the Windows platform (which is my personal computer) rather than the Fedora container on my server to achieve a higher accuracy during the test-train-split.  

I don't know if that has anything to do with the actual operating system and being less ambiguous on the resources by running directly in Linux compared to Windows, or if it is the actual computational power of the server achieving better results with the xenon processors running with 24 cores.  

Just a fun little observation.
